kratos/geometries/line_2d_2.h

```diff
@@ -1054,12 +1054,57 @@ class Line2D2 : public Geometry<TPointType>
     }
 
     ///@}
-    ///@name Friends
+    ///@name Spatial Operations
     ///@{
 
+    /**
+    * @brief Projects a certain point on the geometry, or finds
+    *        the closest point, depending on the provided
+    *        initial guess. The external point does not necessary
+    *        lay on the geometry.
+    *        It shall deal as the interface to the mathematical
+    *        projection function e.g. the Newton-Raphson.
+    *        Thus, the breaking criteria does not necessarily mean
+    *        that it found a point on the surface, if it is really
+    *        the closest if or not. It shows only if the breaking
+    *        criteria, defined by the tolerance is reached.
+    *
+    *        This function requires an initial guess, provided by
+    *        rProjectedPointLocalCoordinates.
+    *        This function can be a very costly operation.
+    *
+    * @param rPointGlobalCoordinates the point to which the
+    *        projection has to be found.
+    * @param rProjectedPointGlobalCoordinates the location of the
+    *        projection in global coordinates.
+    * @param rProjectedPointLocalCoordinates the location of the
+    *        projection in local coordinates.
+    *        The variable is as initial guess!
+    * @param Tolerance accepted of orthogonal error to projection.
+    * @return It is chosen to take an int as output parameter to
+    *         keep more possibilities within the interface.
+    *         0 -> failed
+    *         1 -> converged
+    */
+    int ProjectionPoint(
+        const CoordinatesArrayType& rPointGlobalCoordinates,
+        CoordinatesArrayType& rProjectedPointGlobalCoordinates,
+        CoordinatesArrayType& rProjectedPointLocalCoordinates,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+        ) const override
+    {
+        GeometricalProjectionUtilities::FastProjectOnLine2D(*this, rPointGlobalCoordinates, rProjectedPointGlobalCoordinates);
+
+        PointLocalCoordinates( rProjectedPointLocalCoordinates, rProjectedPointGlobalCoordinates );
+
+        return 1;
```

Comment:
same

Reply:
* @return It is chosen to take an int as output parameter to
    *         keep more possibilities within the interface.
    *         0 -> failed
    *         1 -> converged

Ask @tteschemacher why

Reply:
I'm here to tell you why :)

Some projection functions, as Newton-Raphson return a Boolean to check if the algorithm had converged or if it has reached a termination criteria. This is important to avoid another check from outside.

And well 1 is true and 0 is false.

Reply:
https://tldp.org/LDP/abs/html/exitcodes.html

Exit code 0        Success
Exit code 1        General errors, Miscellaneous errors, such as "divide by zero" and other impermissible operations
Exit code 2        Misuse of shell builtins (according to Bash documentation)        Example: empty_function() {}

Reply:
@tteschemacher but how to tell whether the projection itself is successful or not?

Reply:
But this is not an exit code. This is a convergence check with a mathematical reason. From externally you can just check if true or false.

Reply:
It actually shouldn't fail. You can project any point in space to a line, in a linear operation. In the is inside function you can then check if it's inside.

Reply:
so the intention of this function is only to project and not to tell whether is is a valid projection right?

I.e. one would have to use `IsInside` afterwards to check if it is valid?

Reply:
Yes

Reply:
@tteschemacher ?

Reply:
Yes, exactly.

Is inside tells you if it is inside. ClosestPoint gives you the closest point on the boundary if it is not inside. However, both should ideally use the ProjectPoint.